lib/compress.js

```diff
@@ -2893,7 +2920,7 @@ merge(Compressor.prototype, {
                 });
             }
         }
-        return self;
+        return self.evaluate(compressor)[0];
```

Comment:
@kzc are [these tests](https://github.com/mishoo/UglifyJS2/pull/1425/commits/3f5cdc8bc8df03a4fe68f8d1e242f8203007b883) what you have in mind?